data-scope-main/src/main/resources/static/webview-ui/src/stores/datasource.ts

```
@@ -0,0 +1,406 @@
+      if (ds && ds.metadata && ds.metadata.tables && ds.metadata.tables.length > 0) {
+        return ds.metadata
+      }
+      
+      // 如果没有元数据或元数据为空，则获取单个数据源的详细信息
+      const dataSource = await dataSourceService.getDataSource(dataSourceId)
+      
+      // 如果仍然没有元数据，尝试同步
+      if (!dataSource.metadata || !dataSource.metadata.tables || dataSource.metadata.tables.length === 0) {
+        const syncResult = await syncDataSourceMetadata(dataSourceId)
+        
+        if (!syncResult.success) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)